src/groupeddataframe/groupeddataframe.jl

```diff
@@ -460,6 +462,19 @@ true
 """
 Base.keys(gd::GroupedDataFrame) = GroupKeys(gd)
 
+# assume that GroupKey was generated so that we do not have to check the index
```

Comment:
What does "was generated" mean exactly?

Reply:
"was generated" means that it was produced by DataFrames.jl functions (most likely `keys`).

It is possible to create a `GroupKey` object manually (though it is not exported). Then you can make it point to any index (e.g. very large one that is invalid).

I used this approach because it is also used in `to_index`. But if you prefer (and maybe I prefer it to when you ask 😄) we could also check here if `:idx` field of `GroupKey` is valid and if it is not throw an error (I think we should not return `false` then).

Reply:
I will just  change it also.

Reply:
I guess it wouldn't be a problem for performance, so maybe better also check the index indeed.